lib/sorcery/model/submodules/brute_force_protection.rb

```diff
@@ -63,13 +66,14 @@ def define_brute_force_protection_fields
         module InstanceMethods
           # Called by the controller to increment the failed logins counter.
           # Calls 'login_lock!' if login retries limit was reached.
-          def register_failed_login!
+          def register_failed_login!(password)
             config = sorcery_config
-            return unless login_unlocked?
 
-            sorcery_adapter.increment(config.failed_logins_count_attribute_name)
+            return if login_locked? && !config.limitless_counting_failed_login
 
-            return unless send(config.failed_logins_count_attribute_name) >= config.consecutive_login_retries_amount_limit
+            sorcery_adapter.increment(config.failed_logins_count_attribute_name) unless valid_password?(password)
```

Comment:
I suspect this might open the window to timing attacks, especially considering the trip time to a DB may be significant in certain setups.